crates/wasmtime/src/runtime/module.rs

```diff
@@ -348,8 +414,8 @@ impl Module {
                     &state,
 
                     // Cache miss, compute the actual artifacts
-                    |(engine, wasm)| -> Result<_> {
-                        let (mmap, info) = build_artifacts::<MmapVecWrapper>(engine.0, wasm)?;
+                    |(engine, wasm), dwarf_package| -> Result<_> {
```

Comment:
Instead of modifying `wasmtime-cache`, which is pretty carefully crafted as-is to ensure we don't get cache mistakes, could `dwarf_package` be threaded through with `wasm`?

Reply:
Do you mean a tuple `(engine, wasm, dwarf_package)` or a new struct that has the `wasm` bytes and the dwarf_package `Option<>`?  Sorry I'm new to this so have a few questtions.